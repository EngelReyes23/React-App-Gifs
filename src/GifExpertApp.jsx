import React, { useState } from "react";
import AddCategory from "./components/AddCategory";
import GifGrid from "./components/GifGrid";

const GifExpertApp = ({ defaultCategories }) => {
	//
	const [categories, setCategories] = useState(defaultCategories);

	return (
		<>
			<h1>GifExpertApp</h1>
			<AddCategory setCategories={setCategories} />

			{categories.map((category) => (
				<GifGrid key={category} category={category} />
			))}

			<footer>
				<p>&copy; Engel David Reyes Sandoval</p>
			</footer>
		</>
	);
};

// // PropTypes requeridos
// GifExpertApp.propTypes = {
// 	defaultCategories: PropTypes.array.isRequired,
// };

// PropTypes por default
GifExpertApp.defaultProps = {
	defaultCategories: [],
};

export default GifExpertApp;
