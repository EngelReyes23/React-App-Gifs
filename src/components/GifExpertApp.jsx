import React, { useState } from "react";
import AddCategory from "./AddCategory";
import GifGrid from "./GifGrid";

const GifExpertApp = () => {
	//
	const [categories, setCategories] = useState([]);

	return (
		<>
			<h1>GifExpertApp</h1>
			<AddCategory setCategories={setCategories} />

			{categories.map((category) => (
				<GifGrid key={category} category={category} />
			))}

			<footer>
				<p>&copy; Engel David Reyes Sandoval</p>
			</footer>
		</>
	);
};

export default GifExpertApp;
