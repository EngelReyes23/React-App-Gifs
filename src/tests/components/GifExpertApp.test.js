import GifExpertApp from "../../components/GifExpertApp";
import { shallow } from "enzyme";

describe("Probando el component <GifExpertApp />", () => {
	// 1ra prueba
	test("debe mostrarse correctamente", () => {
		const wrapper = shallow(<GifExpertApp />);
		expect(wrapper).toMatchSnapshot();
	});

	// 2da pruebas
	test("debe mostrar una lista de categorias", () => {
		// arrray de categorias
		const categorias = ["One Punch", "Samurai X", "Dragon Ball"];

		// instacia
		const wrapper = shallow(<GifExpertApp defaultCategories={categorias} />);

		// verificar que se muestre la lista de categorias
		expect(wrapper).toMatchSnapshot();
		expect(wrapper.find("GifGrid").length).toBe(categorias.length);
	});
});
