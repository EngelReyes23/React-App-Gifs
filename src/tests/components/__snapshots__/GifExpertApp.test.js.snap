// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Probando el component <GifExpertApp /> debe mostrar una lista de categorias 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <GifGrid
    category="One Punch"
    key="One Punch"
  />
  <GifGrid
    category="Samurai X"
    key="Samurai X"
  />
  <GifGrid
    category="Dragon Ball"
    key="Dragon Ball"
  />
  <footer>
    <p>
      © Engel David Reyes Sandoval
    </p>
  </footer>
</Fragment>
`;

exports[`Probando el component <GifExpertApp /> debe mostrarse correctamente 1`] = `
<Fragment>
  <h1>
    GifExpertApp
  </h1>
  <AddCategory
    setCategories={[Function]}
  />
  <footer>
    <p>
      © Engel David Reyes Sandoval
    </p>
  </footer>
</Fragment>
`;
